Code/Tools/AssetProcessor/native/utilities/ProductOutputUtil.cpp

```diff
@@ -96,23 +119,73 @@ namespace AssetProcessor
                         }
                     }
 
-                    bool result = AssetUtilities::MoveFileWithTimeout(oldAbsolutePath.c_str(), newAbsolutePath.c_str(), 1);
+                    DoFileRename(oldAbsolutePath, newAbsolutePath);
+                }
+            }
+            else
+            {
+                for (auto& product : products)
+                {
+                    AZStd::string newName;
 
-                    if (!result)
+                    if (!ComputeFinalProductName(GetTempPrefix(sourceAsset.ScanFolderId()), GetFinalPrefix(sourceAsset.ScanFolderId()), product, newName))
                     {
-                        AZ_Error(
-                            "ProductOutputUtil",
-                            false,
-                            "Failed to move product from " AZ_STRING_FORMAT " to " AZ_STRING_FORMAT
-                            ".  See previous log messages for details on failure.",
-                            AZ_STRING_ARG(oldAbsolutePath),
-                            AZ_STRING_ARG(newAbsolutePath));
+                        continue;
                     }
+
+                    AssetUtilities::ProductPath newProductPath(newName, platformIdentifier);
+                    auto [oldAbsolutePath, newAbsolutePath] = GetPaths(product, platformIdentifier, newProductPath);
+
+                    product.m_productFileName = newName;
+
+                    DoFileRename(oldAbsolutePath, newAbsolutePath);
                 }
             }
         }
     }
 
+    bool ProductOutputUtil::ComputeFinalProductName(
+        const AZStd::string& currentPrefix,
+        const AZStd::string& newPrefix,
+        const AssetBuilderSDK::JobProduct& product,
+        AZStd::string& newName)
+    {
+        QString filename = AZ::IO::PathView(product.m_productFileName).Filename().FixedMaxPathString().c_str();
+
+        int prefixPos = filename.indexOf(currentPrefix.c_str());
+
+        if (prefixPos < 0)
+        {
+            AZ_Error(
+                "ProductOutputUtil",
+                false,
+                "Product " AZ_STRING_FORMAT " is expected to be prefixed but was not",
```

Comment:
This error won't mean a lot to most people. Maybe prefix it with Programmer Error like you've done with other errors, or "Asset Processor Internal Error"? Then non-engineers will know to grab an engineer to help, the engineer will find this code and if they're unfamiliar with the prefix system, there's enough context in this function to work backward toward solving the problem.